File: baselibrary/src/main/java/com/shiye/baselibrary/net/loadingView/LoadingFragment.java
package com.shiye.baselibrary.net.loadingView;

import android.app.DialogFragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.shiye.baselibrary.R;


/**
 * Created by issuser on 2018/7/9.
 * 网络请求的Loading动画
 */

public class LoadingFragment extends DialogFragment {
    private View mLoadingView;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(STYLE_NO_FRAME, R.style.Loading_Dialog);
    }

    @Override
    public void onStart() {
        super.onStart();
        //设置宽度顶满屏幕,无左右留白
        DisplayMetrics dm = new DisplayMetrics();
        getActivity().getWindowManager().getDefaultDisplay().getMetrics(dm);
        getDialog().getWindow().setLayout(dm.widthPixels, getDialog().getWindow().getAttributes().height);

    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        if (mLoadingView == null){
            mLoadingView = inflater.inflate(R.layout.loading_dialog, container, false);
        }
        return mLoadingView;
    }

    public void setLoadingView(View loadingView){
        mLoadingView = loadingView;
    }
}

File: baselibrary/src/main/java/com/shiye/baselibrary/net/loadingView/LoadingFragment.kt
package com.shiye.baselibrary.net.loadingView

import android.app.DialogFragment
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.shiye.baselibrary.R


/**
 * Created by issuser on 2018/7/9.
 * 网络请求的Loading动画
 */

class LoadingFragment : DialogFragment() {
    private var mLoadingView: View? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(DialogFragment.STYLE_NO_FRAME, R.style.Loading_Dialog)
    }

    override fun onStart() {
        super.onStart()
        //设置宽度顶满屏幕,无左右留白
        val dm = DisplayMetrics()
        activity?.let {
            it.windowManager.defaultDisplay.getMetrics(dm)
        }
        dialog.window?.let {
            it.setLayout(dm.widthPixels, it.attributes.height)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle): View? {
        if (mLoadingView == null) {
            mLoadingView = inflater.inflate(R.layout.loading_dialog, container, false)
        }
        return mLoadingView
    }

    fun setLoadingView(loadingView: View?) {
        mLoadingView = loadingView
    }
}
